patches/tModLoader/Terraria.ID/ProjectileID.cs.patch

```diff
@@ -0,0 +1,41 @@
+--- src/Terraria\Terraria.ID\ProjectileID.cs
++++ src/tModLoader\Terraria.ID\ProjectileID.cs
+@@ -1,3 +_,4 @@
++using ReLogic.Reflection;
+ using System;
+ 
+ namespace Terraria.ID
+@@ -1199,5 +_,32 @@
+ 		public const short BookStaffShot = 712;
+ 		public const short DD2Win = 713;
+ 		public const short Count = 714;
++
++		public static readonly IdDictionary Search = IdDictionary.Create<ProjectileID, short>();
++
++		public static string GetUniqueId(int projType) {
++			if ( ProjectileID.Search.ContainsId(projType) ) {
++				return "Terraria " + ProjectileID.Search.GetName(projType);
++			}
++			else {
++				var proj = new Projectile();
++				proj.SetDefaults(projType);
++
++				if ( proj.modProjectile != null ) {
++					return proj.modProjectile.mod.Name + " " + proj.modProjectile.Name;
++				}
++			}
++
++			return "" + projType;
```

Comment:
Please use `thething.ToString()`

Reply:
In theory, the fallthrough should never occur. Maybe I'll just throw an `Exception` instead.

Reply:
Or maybe I'll return `null` instead? I hate doing that with functions that expect a string, in particular.

Reply:
It doesn't matter,it's not about style. .ToString() is conveying and clear, moreso than adding it to an empty string. In c# you replace "" with string.Empty (in Java you don't). If it's a flow that shouldn't happen, then yes, an exception may be preferred.